aten/src/THC/THCScanUtils.cuh

```diff
@@ -207,7 +213,7 @@ __device__ void exclusiveBinaryPrefixScan(T* smem, bool in, T* out, T* carry, Bi
   *out -= (T) in;
 
   // The outgoing carry for all threads is the last warp's sum
-  *carry = smem[(blockDim.x / SCAN_UTILS_WARP_SIZE) - 1];
+  *carry = smem[THCCeilDiv<int>(blockDim.x, SCAN_UTILS_WARP_SIZE) - 1];
```

Comment:
What's going on here? This is CUDA code, and it certainly looks like different semantics to me. Is there a test case that this fixes?

Reply:
OK... but if `SCAN_UTILS_WARP_SIZE == 4` and `blockDim.x == 6`, previously we indexed 0 in smem, and with your change we index 1. Are you sure smem is big enough?

Reply:
smem is passed in from the caller. In case of gatherTopK, yes, I think smem is big enough: https://github.com/pytorch/pytorch/blob/b6f0860c8cb156c4c135ba531db1593e42842856/aten/src/THC/THCTensorTopK.cuh#L377

More importantly, the current logic doesn't seem to follow the intent of the comment. The "last warp" will be in `ceil(blockDim.x,WARP_SIZE)`, as this is how the smem is populated by `inclusiveBinaryPrefixScan`:  https://github.com/pytorch/pytorch/blob/b6f0860c8cb156c4c135ba531db1593e42842856/aten/src/THC/THCScanUtils.cuh#L172

Reply:
OK... so if this actually fixed the problem, can we get the topk tests enabled?

Reply:
Yes, we already have done that as part of a bigger sweep after moving the base compiler to LLVM7. I'll prepare a separate PR for that tomorrow.

Reply:
OK, after finally sitting down and working through the indexing, I now agree the new code is correct.

> Lets say we have `blockDim.x == 9` and `SCAN_UTILS_WARP_SIZE == 4`. The quoted code will write into 0, 0, 0, 0, 1, 1, 1, 1, 2. So we must compute 2 as the last warp. ceil(9, 4) == 3, subtract 1 to get 2, the correct number.